### ISSUE TITLE
Desktop notifications are sent only sporadically

### ISSUE THREAD

**Issue by frdmn** [RocketChat/Rocket.Chat.Electron#1579]
## My Setup

- Operating System: macOS 10.15.4
- App Version: 2.17.9
- Installation type: DMG

- [X] I have tested with the latest version
- [X] I can simulate the issue easily

## Description

Since a few weeks, desktop notifications are not realiable anymore on my 3.1.1 (updated today, before it was 3.0.4 which was affected as well). instance with the Electron client (works fine in browser). If I use the "Test Desktop Notifications" button in the profile, browser console shows:

![Bildschirmfoto 2020-04-15 um 14 29 58](https://user-images.githubusercontent.com/944459/79337343-9e070880-7f25-11ea-8b95-5e1986f5a1e1.png)

<details><summary>Plaintext version</summary>
<p>

```
notifications.js:77 Uncaught (in promise) TypeError: Only absolute URLs are supported
    at getNodeRequestOptions (/Applications/Rocket.Chat.New.app/Contents/Resources/app.asar/node_modules/electron-fetch/lib/index.js:1220)
    at /Applications/Rocket.Chat.New.app/Contents/Resources/app.asar/node_modules/electron-fetch/lib/index.js:1293
    at new Promise (<anonymous>)
    at /Applications/Rocket.Chat.New.app/Contents/Resources/app.asar/node_modules/electron-fetch/lib/index.js:1290
(anonymous) @ remote.js:265
metaToPlainObject @ remote.js:265
error @ remote.js:219
metaToValue @ remote.js:225
(anonymous) @ remote.js:216
array @ remote.js:216
metaToValue @ remote.js:225
(anonymous) @ remote.js:286
(anonymous) @ remote.js:276
emit @ events.js:203
EventEmitter.emit @ domain.js:476
onMessage @ init.ts:50
async function (async)
create @ notifications.js:59
Notification @ notifications.js:52
(anonymous) @ 2c035fa7a13a201fc64b2125b64ec902d155340f.js?meteor_js_resource=true:973
(anonymous) @ 2c035fa7a13a201fc64b2125b64ec902d155340f.js?meteor_js_resource=true:1
Promise.then (async)
n.then @ 2c035fa7a13a201fc64b2125b64ec902d155340f.js?meteor_js_resource=true:653
notify @ 2c035fa7a13a201fc64b2125b64ec902d155340f.js?meteor_js_resource=true:973
click .js-test-notifications @ 2c035fa7a13a201fc64b2125b64ec902d155340f.js?meteor_js_resource=true:973
(anonymous) @ 2c035fa7a13a201fc64b2125b64ec902d155340f.js?meteor_js_resource=true:750
t._withTemplateInstanceFunc @ 2c035fa7a13a201fc64b2125b64ec902d155340f.js?meteor_js_resource=true:750
(anonymous) @ 2c035fa7a13a201fc64b2125b64ec902d155340f.js?meteor_js_resource=true:750
(anonymous) @ 2c035fa7a13a201fc64b2125b64ec902d155340f.js?meteor_js_resource=true:750
h._withCurrentView @ 2c035fa7a13a201fc64b2125b64ec902d155340f.js?meteor_js_resource=true:750
(anonymous) @ 2c035fa7a13a201fc64b2125b64ec902d155340f.js?meteor_js_resource=true:750
delegatedHandler @ 2c035fa7a13a201fc64b2125b64ec902d155340f.js?meteor_js_resource=true:750
dispatch @ 2c035fa7a13a201fc64b2125b64ec902d155340f.js?meteor_js_resource=true:517
g.handle @ 2c035fa7a13a201fc64b2125b64ec902d155340f.js?meteor_js_resource=true:517
```

</p>
</details>

### Current Behavior

Desktop notifications not realiable

### Expected Behavior

Desktop notifications should be sent whenever they get triggered.

**Comment by versedi** [comment]
On Ubuntu 19.10 and both 2.16 and 2.17.

I don't get notifications or sounds at all. Very frustrating.

**Comment by thorin8k** [comment]
I'm getting the same errors on Windows so it seems to be a bug in the client behaviour.

Platform Windows 10.
Client Version: 2.17.7

**Comment by klou** [comment]
Same on Windows 10/Client 2.17.9

**Comment by klou** [comment]
https://github.com/RocketChat/Rocket.Chat.Electron/blob/3596a26ffb9ddd1ce8ea4fd466372c97ebbefe85/src/preload/notifications.js#L57-L60

Just a guess, but it is referencing a server icon (i.e. default Rocket.Chat icon) on the Test Desktop Notification (which utilizes electron-fetch), and failing because a local path is being returned?

**Comment by dionatan1989** [comment]
![image](https://user-images.githubusercontent.com/11030322/84779415-c14c4380-afba-11ea-999c-63b4b5b3dc95.png)

Same problem here.

**Comment by dionatan1989** [comment]
![image](https://user-images.githubusercontent.com/11030322/84779692-15efbe80-afbb-11ea-8b4e-7980384d7f67.png)

Same problem here.

**Comment by frdmn** [comment]
Can you guys please try version 2.17.11 which was just released? 

The PR https://github.com/RocketChat/Rocket.Chat.Electron/pull/1662 should fix this problem. 😊

**Comment by versedi** [comment]
Still investigating: seems to work for some users and for some not.

Does this require a server update as well or only the client? 

Is it possible all clients talking with each other need to be updated?

**Comment by tassoevan** [comment]
Closed by #1662
